Fundamentals/Setup/Server-Setup/runtime-modes.md

```diff
@@ -0,0 +1,105 @@
+
+## Configuring the runtime mode
+
+You can configure the runtime mode to optimize Umbraco for different development experiences and environments by setting `Umbraco:CMS:Runtime:Mode` to one of the available modes:
+
+- `BackofficeDevelopment` (default)
+- `Development`
+- `Production`
+
+This can be done via the `appsettings.json` file, environment variables or any other .NET configuration provider (like Azure Key Vault/App Configuration). Although this setting affects how Umbraco behaves at runtime, some modes have prerequisites on how the project is built/published. Make sure to read the descriptions of each mode before changing this setting from the default `BackofficeDevelopment` mode, as incorrect configuration can result in your application not starting (by throwing a `BootFailedException`).
+
+## BackofficeDevelopment mode
+
+The `BackofficeDevelopment` mode is the default behavior for Umbraco: it does not optimize Umbraco for any specific environment and does not have any prerequisites. This mode allows for rapid development (without having to recompile/rebuild your project), including doing all development from within the backoffice.
+
+## Development mode
+
+The `Development` mode can be used when you're developing from an IDE (like Visual Studio, VS Code or Rider) or the dotnet CLI (e.g. using `dotnet watch`) and is a recommend prerequisite if you want to use `Production` mode in your production environment.
+
+This mode disables in-memory ModelsBuilder generation and validates the following setting:
+- `Umbraco:CMS:ModelsBuilder:ModelsMode` is not set to `InMemoryAuto`.
+
+If you want to use the generated models, use `SourceCodeAuto` or `SourceCodeManual`, which requires manually recompiling the project after the models have changed (e.g. after updating Document Types, Media Types, Member Types or Data Types). Razor views (`cshtml` files) will still be automatically compiled at runtime, allowing you to quickly iterate on the rendered output from templates, (macro) partial views and view components.
+
+The recommended approach to enable `Development` mode is to update the `appsettings.json` file with the following settings:
+
+```json
+{
+    "Umbraco": {
+        "CMS": {
+            "Runtime": {
+                "Mode" : "Development"
+            },
+            "ModelsBuilder":{
+                "ModelsMode": "SourceCodeAuto"
+            }
+        }
+    }
+}
+```
+
+Now ensure your models are generated by running Umbraco and going to Settings > Models Builder > Generate models. You can now remove the following properties from your `csproj` project file to enable compilation of Razor views (which also ensures your views don't contain compilation errors and is a prerequisite for enabling `Production` mode):
+
+```xml
+<RazorCompileOnBuild>false</RazorCompileOnBuild>
+<RazorCompileOnPublish>false</RazorCompileOnPublish>
+<CopyRazorGenerateFilesToPublishDirectory>true</CopyRazorGenerateFilesToPublishDirectory>
```

Comment:
After some additional testing, the `CopyRazorGenerateFilesToPublishDirectory` property needs to be left enabled to ensure Deploy can generate signatures for templates (otherwise you'll get schema mismatches):
```suggestion
```